File: fastapi_back/app/models/attachment_folder.py
"""
附件文件夹模型
"""
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional, List, TYPE_CHECKING
from datetime import datetime

if TYPE_CHECKING:
    from app.models.project import Project
    from app.models.historical_project import HistoricalProject
    from app.models.attachment import Attachment


class AttachmentFolderBase(SQLModel):
    project_id: Optional[int] = Field(default=None, foreign_key="project.id", description="所属项目ID")
    historical_project_id: Optional[int] = Field(default=None, foreign_key="historicalproject.id", description="所属历史项目ID")
    name: str = Field(index=True, description="文件夹名称")
    is_default: bool = Field(default=False, description="是否为默认文件夹")
    description: Optional[str] = None


class AttachmentFolder(AttachmentFolderBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow, sa_column_kwargs={"onupdate": datetime.utcnow})
    
    # 关系
    project: Optional["Project"] = Relationship(back_populates="folders")
    historical_project: Optional["HistoricalProject"] = Relationship(back_populates="folders")
    attachments: List["Attachment"] = Relationship(back_populates="folder", sa_relationship_kwargs={"cascade": "all, delete-orphan"})


class AttachmentFolderCreate(SQLModel):
    name: str
    description: Optional[str] = None


class AttachmentFolderUpdate(SQLModel):
    name: Optional[str] = None
    description: Optional[str] = None


class AttachmentFolderRead(AttachmentFolderBase):
    id: int
    created_at: datetime
    updated_at: datetime
    attachment_count: Optional[int] = 0  # 附件数量

File: project_manager_vue3/src/api/attachmentFolder.ts
import request from './request'

export interface AttachmentFolder {
  id: number
  project_id: number
  name: string
  is_default: boolean
  description?: string
  created_at: string
  updated_at: string
  attachment_count?: number
}

export interface AttachmentFolderCreate {
  name: string
  description?: string
}

export interface AttachmentFolderUpdate {
  name?: string
  description?: string
}

export const attachmentFolderApi = {
  list: (projectId: number) => {
    return request.get<AttachmentFolder[]>(`/attachment-folders/project/${projectId}`)
  },
  
  create: (projectId: number, data: AttachmentFolderCreate) => {
    return request.post<AttachmentFolder>(`/attachment-folders/project/${projectId}`, data)
  },
  
  update: (folderId: number, data: AttachmentFolderUpdate) => {
    return request.put<AttachmentFolder>(`/attachment-folders/${folderId}`, data)
  },
  
  delete: (folderId: number) => {
    return request.delete(`/attachment-folders/${folderId}`)
  },
}

